### PR TITLE
Make source paths configured per source repository (breaking)

### PR DESCRIPTION
**Pull request by marians**
Closes https://github.com/giantswarm/crd-docs-generator/issues/91

**This is a breaking change. Configuration files have to be adapted to work with this version. In contrast to previous versions, there are no more default paths to search for CRDs, example CRs, or annotations.**

Source paths for CRDs, example CRs, annotations are now configured in the config file.

Three new configuration keys are introduced to configure paths per `source_repository`:

- `crd_paths`: paths to search for CRD YAML files.
- `cr_paths`: paths to search for example CR YAML files.
- `annotations_paths`: paths to search for Go files defining annotations.

All expect array values. Paths are relative to the source repo root. See the `config.example.yaml` file to learn how to use the keys.

## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: main.go
Comment (marians) [inline main.go:124]: The differentiation between own and upstream CRDs wasn't necessary and could be removed.

**Review by anvddriesch — APPROVED** [review]

**Comment by jmazzitelli** [comment]
@marians I can confirm this works. Very nice!

Here's what I tested with - https://github.com/jmazzitelli/kiali-operator/blob/crd-gen-fix-root-dir-2/crd-docs/config/apigen-config.yaml

File: pkg/config/config_test.go
Comment (tfussell) [inline pkg/config/config_test.go:30]: Everything in `config/crd` should be copied into `helm/*/templates` so it should be safe to delete `config/crd`.

**Review by tfussell — APPROVED** [review]

File: pkg/config/config_test.go
Comment (marians) [reply pkg/config/config_test.go:30]: FYI: This test only verifies that we read the configuration file okay.

**Comment by jmazzitelli** [comment]
@marians @tfussell @anvddriesch 

Thanks for the quick enhancement!

Would you be able to provide an ETA for the [container image release on quay](https://quay.io/repository/giantswarm/crd-docs-generator?tab=tags) that contains this new feature? I do not know the cadence of your release cycles. If it will be soon, I will wait before I merge my [project's PR](https://github.com/kiali/kiali-operator/pull/470) so I can use this new feature now. If it will not be released soon, I will merge my PR now, and when the new crd-docs-generator release is available, I can submit a followup PR on my project.

Thanks again.

**Comment by marians** [comment]
The image `quay.io/giantswarm/crd-docs-generator:0.9.0` is now available. Yesterday a CircleCI outage prevented it from being created, but now I restarted the workflow and everything is fine.

**Pull request merged**
